src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -170,9 +171,40 @@ public async Task<SecretResponseModel> UpdateSecretAsync([FromRoute] Guid id, [F
     [HttpPost("secrets/delete")]
     public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([FromBody] List<Guid> ids)
     {
-        var userId = _userService.GetProperUserId(User).Value;
-        var results = await _deleteSecretCommand.DeleteSecrets(ids, userId);
-        var responses = results.Select(r => new BulkDeleteResponseModel(r.Item1.Id, r.Item2));
+        var secrets = (await _secretRepository.GetManyByIds(ids)).ToList();
+        if (!secrets.Any() || secrets.Count != ids.Count)
+        {
+            throw new NotFoundException();
+        }
+
+        // Ensure all secrets belong to the same organization.
+        var organizationId = secrets.First().OrganizationId;
+        if (secrets.Any(secret => secret.OrganizationId != organizationId) ||
+            !_currentContext.AccessSecretsManager(organizationId))
+        {
+            throw new NotFoundException();
+        }
+
+        var secretsToDelete = new List<Secret>();
+        var results = new List<(Secret Secret, string Error)>();
+
+        foreach (var secret in secrets)
+        {
+            var authorizationResult =
+                await _authorizationService.AuthorizeAsync(User, secret, SecretOperations.Delete);
+            if (authorizationResult.Succeeded)
+            {
+                secretsToDelete.Add(secret);
+                results.Add((secret, ""));
+            }
+            else
+            {
+                results.Add((secret, "access denied"));
+            }
+        }
+
+        await _deleteSecretCommand.DeleteSecrets(secretsToDelete);
```

Comment:
Just want to double check - we for sure want to continue with a bulk delete even if some of them fail? (We don't want to bail?)

Reply:
Since this is just a refactor, I didn't want to modify how the API behaves currently.

So just like current state, this will continue to process even if one is access denied.

If we want to pursue changing the client and server with a different API behavior, I'm open, but I don't think it should be in scope for this effort.

Reply:
Sounds good 👍